modules/ingest-grok/src/main/java/org/elasticsearch/ingest/grok/GrokProcessor.java

```diff
@@ -58,14 +77,44 @@ public String getType() {
         return TYPE;
     }
 
-    String getMatchField() {
+    public String getMatchField() {
         return matchField;
     }
 
-    Grok getGrok() {
+    public Grok getGrok() {
         return grok;
     }
 
+    // TODO(talevy): cleanup with builders? make this a Collector?
+    private static String combinePatterns(List<String> patterns, boolean traceMatch) {
```

Comment:
Can we add a unit test for this method that tests the combining logic in isolation?